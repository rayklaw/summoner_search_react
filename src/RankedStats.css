.RankedStatsSolo, .RankedStatsFlex, .RankedStatsTft {
    display: inline-block;
    background-color: dodgerblue;
    text-align: center;
    height: 129px;
    width: 400px;
    margin-left: 20px
  }

  .RankedStatsSolo > img, .RankedStatsFlex > img, .RankedStatsTft > img{
    float: left;
    padding: 5px;
    width: 120px;
    height: 120px;
    margin: 0px -20px -10px 15px;
  }
  
  
  .RankedStatsSolo > div, .RankedStatsFlex > div, .RankedStatsTft > div {
    background-color: rgb(248, 248, 248);
    text-align: center;
    margin: 2px;
    padding: 10px;
    line-height: 3px;
    font-size: 15px;
  }