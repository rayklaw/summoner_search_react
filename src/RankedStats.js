import React from 'react'

import './RankedStats.css'

const RankedStats = props => {
    const { rankedStats } = props

    let ordered = new Array(3)

    function fromRoman(string) {
        if (string === 'I') { return 1 }
        if (string === 'II') { return 2 }
        if (string === 'III') { return 3 }
        if (string === 'IV') { return 4 }
    }

    rankedStats.forEach((stat, index) => {
        if (stat.queueType === "RANKED_SOLO_5x5") {
            ordered[0] = rankedStats[index]
        }
        if (stat.queueType === "RANKED_FLEX_SR") {
            ordered[1] = rankedStats[index]
        }
        if (stat.queueType === "RANKED_TFT") {
            ordered[2] = rankedStats[index]
        }
    })

    return (
        <div className="RankedStats">
            <div className="RankedStatsSolo">
                {ordered[0] && <img src={`https://opgg-static.akamaized.net/images/medals/${ordered[0].tier.toLowerCase()}_${fromRoman(ordered[0].rank)}.png`} alt="Ranked Tier Icon"></img>}
                {!ordered[0] && <img src='http://opgg-static.akamaized.net/images/medals/default.png' alt="Ranked Tier Icon"></img>}
                <div>
                    <p>Ranked Solo Queue</p>
                    {ordered[0] && <p>{ordered[0].tier} {ordered[0].rank} {ordered[0].leaguePoints} LP</p>}<br></br>
                    {ordered[0] && <p>{ordered[0].wins} Wins / {ordered[0].losses} Losses</p>}
                    {ordered[0] && <p>{Math.floor(ordered[0].wins / (ordered[0].wins + ordered[0].losses) * 100)}% Winrate</p>}
                    {!ordered[0] && <p>Unranked</p>}
                </div>
            </div>
            <div className="RankedStatsFlex">
                {ordered[1] && <img src={`https://opgg-static.akamaized.net/images/medals/${ordered[1].tier.toLowerCase()}_${fromRoman(ordered[1].rank)}.png`} alt="Ranked Tier Icon"></img>}
                {!ordered[1] && <img src='http://opgg-static.akamaized.net/images/medals/default.png' alt="Ranked Tier Icon"></img>}
                <div>
                    <p>Ranked Flex Queue</p>
                    {ordered[1] && <p>{ordered[1].tier} {ordered[1].rank} {ordered[1].leaguePoints} LP</p>}<br></br>
                    {ordered[1] && <p>{ordered[1].wins} Wins / {ordered[1].losses} Losses</p>}
                    {ordered[1] && <p>{Math.floor(ordered[1].wins / (ordered[1].wins + ordered[1].losses) * 100)}% Winrate</p>}
                    {!ordered[1] && <p>Unranked</p>}
                </div>
            </div>
            <div className="RankedStatsTft">
                {ordered[2] && <img src={`https://opgg-static.akamaized.net/images/medals/${ordered[2].tier.toLowerCase()}_${fromRoman(ordered[2].rank)}.png`} alt="Ranked Tier Icon"></img>}
                {!ordered[2] && <img src='http://opgg-static.akamaized.net/images/medals/default.png' alt="Ranked Tier Icon"></img>}
                <div>
                    <p>Ranked Teamfight Tactics</p>
                    {ordered[2] && <p>{ordered[2].tier} {ordered[2].rank} {ordered[2].leaguePoints} LP</p>}<br></br>
                    {ordered[2] && <p>{ordered[2].wins} Wins / {ordered[2].losses} Losses</p>}
                    {ordered[2] && <p>{Math.floor(ordered[2].wins / (ordered[2].wins + ordered[2].losses) * 100)}% Winrate</p>}
                    {!ordered[2] && <p>Unranked</p>}
                </div>
            </div>
        </div>
    )
}

export default RankedStats