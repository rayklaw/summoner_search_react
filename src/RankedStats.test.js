import React from 'react'
import ReactDOM from 'react-dom'
import TestRenderer from 'react-test-renderer'
import ReactTestUtils from 'react-dom/test-utils'

import sinon from 'sinon'

import RankedStats from './RankedStats'

import * as mock from './mock'

it('ranked stats renders', () => {
    let rankedStats = [{
        "leagueId": "d42df200-3018-11e9-9bdb-c81f66cf135e",
        "queueType": "RANKED_SOLO_5x5",
        "tier": "CHALLENGER",
        "rank": "I",
        "summonerId": "NtW1fFWiqG2Nqw5fo1NaUeI4ouvXXcHFT4hz6VsrEy5eSsE",
        "summonerName": "MyAir",
        "leaguePoints": 64,
        "wins": 4,
        "losses": 20,
        "veteran": false,
        "inactive": false,
        "freshBlood": false,
        "hotStreak": false
    }]

    const component = TestRenderer.create(<RankedStats rankedStats={rankedStats} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})