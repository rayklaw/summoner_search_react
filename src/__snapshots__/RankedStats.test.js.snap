// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ranked stats renders 1`] = `
<div
  className="RankedStats"
>
  <div
    className="RankedStatsSolo"
  >
    <img
      alt="Ranked Tier Icon"
      src="https://opgg-static.akamaized.net/images/medals/challenger_1.png"
    />
    <div>
      <p>
        Ranked Solo Queue
      </p>
      <p>
        CHALLENGER
         
        I
         
        64
         LP
      </p>
      <br />
      <p>
        4
         Wins / 
        20
         Losses
      </p>
      <p>
        16
        % Winrate
      </p>
    </div>
  </div>
  <div
    className="RankedStatsFlex"
  >
    <img
      alt="Ranked Tier Icon"
      src="http://opgg-static.akamaized.net/images/medals/default.png"
    />
    <div>
      <p>
        Ranked Flex Queue
      </p>
      <br />
      <p>
        Unranked
      </p>
    </div>
  </div>
  <div
    className="RankedStatsTft"
  >
    <img
      alt="Ranked Tier Icon"
      src="http://opgg-static.akamaized.net/images/medals/default.png"
    />
    <div>
      <p>
        Ranked Teamfight Tactics
      </p>
      <br />
      <p>
        Unranked
      </p>
    </div>
  </div>
</div>
`;
